src/ansys/fluent/core/solver/error_message.py

```diff
@@ -45,14 +45,15 @@ def allowed_name_error_message(
     if not message:
         message = f"'{context}' has no attribute '{trial_name}'"
     message += ".\n"
-    matches = None
     if allowed_values:
-        if isinstance(allowed_values, list) and isinstance(allowed_values[0], str):
+        matches = None
+        if all(isinstance(item, str) for item in allowed_values) and allowed_values:
```

Comment:
@seanpearsonuk, we have to put in the 'str' check as of now because the field_data is also using this with the surface ids as allowed values which are int. We need to update these methods in a separate PR to correct the error message as well.

Reply:
I have updated the check to test for a homogenous 'str' list.

Reply:
That's nice, thanks, @prmukherj.

Reply:
Just wondering:
1. Why is the caller passing in a `str` `name` with an `Iterable[int]` to constrain that name. How is `Iterable[int]` used if at all? A bit confused by this.
2. The annotation shows `Iterable[str | int]`. Does the code require that specifically? It seems that it's either `Iterable[str]` or `Iterable[Any]` depending on the outcome of pt. 1 above.

Reply:
1. Actually the closest match is required if the allowed values and the entered name is str, however there is one particular case where user passes surface id list (int) and in this case only allowed values is displayed.
2. Right, actually here it should be Iterable[str] | Iterable[int]

Reply:
Updated the type-hints.